src/html2pdf.js

```diff
@@ -32,356 +32,412 @@
+            var pageCanvas = document.createElement('canvas');
+            var pageCtx = pageCanvas.getContext('2d');
+            var pageHeight = pageSize.inner.height;
+            pageCanvas.width = canvas.width;
+            pageCanvas.height = pxPageHeight;
+
+            // Initialize the PDF.
+            var pdf = new jsPDF(opt.jsPDF);
+
+            /**
+             * added support for page split only on white "rows".
+             * check if pixel is not "white" if it is less than whiteColorLevel value for the rgb values
+             * if the row isn't "white" - will go one pixel up and test that row, up to max. number of maxPixelToGoUp rows up
+             *
+             */
+            var whiteColorLevel = 150;
+            var maxPixelToGoUp = 150;
+            var pageStartPos = 0;
+            for (var page = 0; page < nPages; page++) {
+                // Trim the final page to reduce file size.
+                if (page === nPages - 1) {
+                    pageCanvas.height = pageModulus;
+                    pageHeight = pageCanvas.height * pageSize.inner.width / pageCanvas.width;
+                }
+                // Display the page.
+                var w = pageCanvas.width;
+                var heightDrawBack = 0;
+                do {
+
+                    var h = pageCanvas.height - heightDrawBack;
+                    var whiteRow = true;
+                    var wPix = 0;
+                    var bPixCount = 0;
+
+                    while (wPix < w && whiteRow) {
+                        var pixelData = ctx.getImageData(wPix, pageStartPos + h, 1, 1).data;
+                        if (pixelData[0] < whiteColorLevel || pixelData[1] < whiteColorLevel || pixelData[2] < whiteColorLevel) {
+                            bPixCount++;
+                            if (bPixCount > 1) {
+                                whiteRow = false;
+                            }
+                        }
+                        wPix++;
+                    }
+                    heightDrawBack++;
+                }
+                while (!whiteRow && heightDrawBack < maxPixelToGoUp) ;
+                heightDrawBack--;
+                pageModulus += heightDrawBack;
+                if (pageModulus >= pxPageHeight) {
+                    nPages++;
+                    pageModulus -= pxPageHeight;
+                }
+                pageCtx.fillStyle = 'white';
+                pageCtx.fillRect(0, 0, w, h);
+                pageCtx.drawImage(canvas, 0, pageStartPos, w, h, 0, 0, w, h);
+
+                if (heightDrawBack) pageCtx.fillRect(0, h, w, heightDrawBack);
+
+                // Add the page to the PDF.
+                if (page) pdf.addPage();
+                var imgData = pageCanvas.toDataURL('image/' + opt.image.type, opt.image.quality);
+                pdf.addImage(imgData, opt.image.type, opt.margin[1], opt.margin[0],
+                    pageSize.inner.width, pageHeight);
```

Comment:
It would be worth altering `pageHeight` so that only the top (non-white) portion of the image is copied to the PDF, to reduce file size (same reason the very last page is trimmed).

Reply:
agreed